lib/node/nodes/intersection.js

```diff
@@ -1,28 +1,41 @@
 'use strict';
 
 var Node = require('../node');
+var debug = require('../../util/debug')('analysis:intersection');
 
 var TYPE = 'intersection';
 var PARAMS = {
     source: Node.PARAM.NODE(Node.GEOMETRY.ANY),
-    target: Node.PARAM.NODE(Node.GEOMETRY.ANY)
+    source_columns: Node.PARAM.NULLABLE(Node.PARAM.ARRAY(Node.PARAM.STRING()), []),
+    target: Node.PARAM.NODE(Node.GEOMETRY.ANY),
 };
 
-var Intersection = Node.create(TYPE, PARAMS, { cache: true, version: 3 });
+var Intersection = Node.create(TYPE, PARAMS, {
+    cache: true,
+    version: 4,
+    beforeCreate: function (node) {
```

Comment:
But you can do that outside of `beforeCreate`. The `beforeCreate` is supposed to be used when you cannot do something after the node has been created: ignore param for id generation, throw an exception due a specific validation, and similar things.